Upstream-Status: Inappropriate [backport from upstream]

In May, people already reported the mesa-dri build issue when upgrading to
glproto-1.4.13 and dri2proto-2.4:
http://lists.freedesktop.org/archives/mesa-dev/2011-May/007331.html

The story is: Jesse tried to change the definitions of some structs in
glproto, dri2proto and mesa-dri, but the change in mesa-dri side was accepted
and then reverted, see
http://cgit.freedesktop.org/mesa/mesa/commit/?id=dc4a3edcf9d414d5098618d5d3b79d8dd90adcc3

so in glproto and dri2proto's sides, we have to revert the change, too:
http://cgit.freedesktop.org/xorg/proto/glproto/commit/?id=7196c9441af941be74180f8e8d4e10a08659b80f

We could remove this patch when upgrading to newer glproto, dri2proto and
mesa-dri in future.

Wed Jun 15 14:33:14 CST 2011
Dexuan Cui <dexuan.cui@intel.com>

-------------------------------------------------------

commit 7196c9441af941be74180f8e8d4e10a08659b80f
Author: Jesse Barnes <jbarnes@virtuousgeek.org>
Date:   Thu May 5 09:21:19 2011 -0700

    Revert "glxproto: make GLX swap event struct match spec"
    
    This reverts commit f5403828c68af0f12c79c0504df0a4781ca84b2b.
    
    This broke compatibility with existing code, so revert in favor of a new
    struct definition instead.

diff --git a/glxproto.h b/glxproto.h
index dfa0647..0ff44e3 100644
--- a/glxproto.h
+++ b/glxproto.h
@@ -1375,13 +1375,13 @@ typedef struct {
     BYTE pad;
     CARD16 sequenceNumber B16;
     CARD16 event_type B16;
-    CARD16 pad2;
     CARD32 drawable;
     CARD32 ust_hi B32;
     CARD32 ust_lo B32;
     CARD32 msc_hi B32;
     CARD32 msc_lo B32;
-    CARD32 sbc B32;
+    CARD32 sbc_hi B32;
+    CARD32 sbc_lo B32;
 } xGLXBufferSwapComplete;
 
 /************************************************************************/
